test/parquet_schema_test.cc

```diff
@@ -17,50 +17,494 @@
+  return ::parquet::schema::GroupNode::Make(
+      name, optional ? ::parquet::Repetition::OPTIONAL : ::parquet::Repetition::REQUIRED,
+      fields, /*logical_type=*/nullptr, field_id);
 }
 
+::parquet::schema::NodePtr MakeListNode(const std::string& name,
+                                        const ::parquet::schema::NodePtr& element_node,
+                                        int field_id = -1, bool optional = true) {
+  auto list_group = ::parquet::schema::GroupNode::Make(
+      "element", ::parquet::Repetition::REPEATED, {element_node});
+  return ::parquet::schema::GroupNode::Make(
+      name, optional ? ::parquet::Repetition::OPTIONAL : ::parquet::Repetition::REQUIRED,
+      {list_group}, ::parquet::LogicalType::List(), field_id);
+}
+
+::parquet::schema::NodePtr MakeMapNode(const std::string& name,
+                                       const ::parquet::schema::NodePtr& key_node,
+                                       const ::parquet::schema::NodePtr& value_node,
+                                       int field_id = -1, bool optional = true) {
+  auto key_value_group = ::parquet::schema::GroupNode::Make(
+      "key_value", ::parquet::Repetition::REPEATED, {key_node, value_node});
+  return ::parquet::schema::GroupNode::Make(
+      name, optional ? ::parquet::Repetition::OPTIONAL : ::parquet::Repetition::REQUIRED,
+      {key_value_group}, ::parquet::LogicalType::Map(), field_id);
+}
+
+// Helper to create SchemaManifest from Parquet schema
+::parquet::arrow::SchemaManifest MakeSchemaManifest(
+    const ::parquet::schema::NodePtr& parquet_schema) {
+  auto parquet_schema_descriptor = std::make_shared<::parquet::SchemaDescriptor>();
+  parquet_schema_descriptor->Init(parquet_schema);
+
+  ::parquet::arrow::SchemaManifest manifest;
+  auto status = ::parquet::arrow::SchemaManifest::Make(
+      parquet_schema_descriptor.get(), /*key_value_metadata=*/nullptr,
+      ::parquet::default_arrow_reader_properties(), &manifest);
+  if (!status.ok()) {
+    throw std::runtime_error("Failed to create SchemaManifest: " + status.ToString());
+  }
+  return manifest;
+}
+
+#define ASSERT_PROJECTED_FIELD(field_projection, index)                \
+  ASSERT_EQ(field_projection.kind, FieldProjection::Kind::kProjected); \
+  ASSERT_EQ(std::get<1>(field_projection.from), index);
+
+#define ASSERT_PROJECTED_NULL_FIELD(field_projection) \
+  ASSERT_EQ(field_projection.kind, FieldProjection::Kind::kNull);
+
 }  // namespace
 
-TEST(HasFieldIds, PrimitiveNode) {
+TEST(HasFieldIdsTest, PrimitiveNode) {
   EXPECT_FALSE(HasFieldIds(MakeInt32Node("test_field")));
   EXPECT_TRUE(HasFieldIds(MakeInt32Node("test_field", /*field_id=*/1)));
 }
 
-TEST(HasFieldIds, GroupNode) {
-  auto group_node_without_field_id =
-      MakeGroupNode("test_group", {MakeInt32Node("c1"), MakeInt32Node("c2")});
-  EXPECT_FALSE(HasFieldIds(group_node_without_field_id));
+TEST(HasFieldIdsTest, GroupNode) {
```

Comment:
I guess the original test name `HasFieldIds` conflicts with the function name `HasFieldIds` so the `TEST` expansion may have confused the linter. @lidavidm @mapleFU